code/processes/wdb.q

```diff
@@ -102,149 +107,212 @@ mergemaxrows:{[tabname] mergenumrows^mergenumtab[tabname]}
 
 /- function to return a list of tables that the wdb process has been configured to deal within
 tablelist:{[] sortedlist:exec tablename from `bytes xdesc .wdb.tabsizes;
-	(sortedlist union tables[`.]) except ignorelist}
+    (sortedlist union tables[`.]) except ignorelist}
 
 /- function to upsert to specified directory
-upserttopartition:{[dir;tablename;tabdata;pt;expttype;expt]	    		
-	.lg.o[`save;"saving ",(string tablename)," data to partition ",
-		/- create directory location for selected partiton
-		string directory:` sv .Q.par[dir;pt;tablename],
-		/- replace random chracters in symbols with _
-		(`$"_"^.Q.an .Q.an?"_" sv string 
-		/- convert to symbols and replace any null values with `NONE
-		`NONE^ -1 _ `${@[x; where not ((type each x) in (10 -10h));string]} expt,(::)),`];	
-	/- upsert selected data matched on partition to specific directory 	
-	.[
-		upsert;
-		(directory;r:?[tabdata;{(x;y;(),z)}[in;;]'[expttype;expt];0b;()]);		
-		{[e] .lg.e[`savetablesbypart;"Failed to save table to disk : ",e];'e}
-	];
-	.lg.o[`track;"appending details to partsizes"];
-	/-key in partsizes are directory to partition, need to drop training slash in directory key
-	.merge.partsizes[first ` vs directory]+:(count r;-22!r);
-	};
-	
+upserttopartition:{[dir;tablename;tabdata;pt;expttype;expt]
+    .lg.o[`save;"saving ",(string tablename)," data to partition ",
+        /- create directory location for selected partiton
+        string directory:` sv .Q.par[dir;pt;tablename],
+        /- replace random chracters in symbols with _
+        (`$"_"^.Q.an .Q.an?"_" sv string
+        /- convert to symbols and replace any null values with `NONE
+        `NONE^ -1 _ `${@[x; where not ((type each x) in (10 -10h));string]} expt,(::)),`];
+    /- upsert selected data matched on partition to specific directory
+    .[
+        upsert;
+        (directory;r:?[tabdata;{(x;y;(),z)}[in;;]'[expttype;expt];0b;()]);
+        {[e] .lg.e[`savetablesbypart;"Failed to save table to disk : ",e];'e}
+    ];
+    .lg.o[`track;"appending details to partsizes"];
+    /-key in partsizes are directory to partition, need to drop training slash in directory key
+    .merge.partsizes[first ` vs directory]+:(count r;-22!r);
+    };
+
+/- function to upsert to specified directory using enumerated extra partitioning
+upserttopartitionenum:{[dir;tablename;tabdata;pt;expttype;expt]
+    hdbsym:` sv hdbsettings[`hdbdir],`sym;
+    /- enumerate current extra partition agains the hdb sym file
+    i:get[hdbsym]?hdbsym?first expt;
```

Comment:
```suggestion
    i:`long$hdbsym?first expt;
```

is this the same thing? (I'm not 100% sure what the intent is here, if it's to get the integer value of the enumerated sym, I think this should be the same & possibly more efficient as we don't have to `get` the sym file)

or possibly we've already enumerated this data, in which case even simpler I think

```suggestion
    i:`long$first expt;
```